qiskit/transpiler/passes/layout/vf2_utils.py

```diff
@@ -95,33 +95,48 @@ def _visit(dag, weight, wire_map):
     return im_graph, im_graph_node_map, reverse_im_graph_node_map, free_nodes
 
 
+def build_edge_list(im_graph):
+    """Generate an edge list for scoring."""
+    return {
+        (edge[0], edge[1]): sum(edge[2].values()) for edge in im_graph.edge_index_map().values()
```

Comment:
We're still eating a `dict` ->` IndexMap` copy/conversion overhead on each scoring. I'll wrap this in a pyclass so we can avoid that overhead too.

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/11115/commits/6c9b34e6876bb698ad29cd55c75f3e39dcd9e5b8